activerecord/test/cases/finder_test.rb

```diff
@@ -436,16 +436,18 @@ def test_last_with_integer_and_order_should_keep_the_order
     assert_equal Topic.order("title").to_a.last(2), Topic.order("title").last(2)
   end
 
-  def test_last_with_integer_and_order_should_not_use_sql_limit
-    query = assert_sql { Topic.order("title").last(5).entries }
+  def test_last_with_integer_and_order_should_use_sql_limit
+    relation = Topic.order("title")
+    query = assert_sql { relation.last(5).entries }
     assert_equal 1, query.length
-    assert_no_match(/LIMIT/, query.first)
+    assert !relation.loaded?
+    assert_match(/LIMIT/, query.first)
```

Comment:
I don't think we need this assertion, if the intent is to make sure that we don't load the entire relation into memory